compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -843,6 +904,18 @@ public void visit(BLangListConstructorExpr.BLangArrayLiteral arrayLiteral) {
 
     @Override
     public void visit(BLangRecordLiteral recordLiteral) {
+        for (RecordLiteralNode.RecordField field : recordLiteral.fields) {
+            if (field.isKeyValueField()) {
+                BLangRecordLiteral.BLangRecordKeyValueField keyValueField =
+                                                                    (BLangRecordLiteral.BLangRecordKeyValueField) field;
+                keyValueField.key.expr = rewriteExpr(keyValueField.key.expr);
+                keyValueField.valueExpr = rewriteExpr(keyValueField.valueExpr);
+            } else if (field.getKind() != NodeKind.SIMPLE_VARIABLE_REF) {
```

Comment:
Why don't we have to handle variable reference fields?

Reply:
Updated(https://github.com/ballerina-platform/ballerina-lang/pull/39349)